### DISCUSSION TITLE
How to import trajectory information saved in the same .pdb or .cif file into blender and restore the real trajectory animation?

### DISCUSSION THREAD

**Discussion by lukun223344** [BradyAJohnston/MolecularNodes#384] [General]
hello,Thank you so much for the plugin, it has made my work so much easier!
Recently, I tried to use AI software to generate some proteins, their generated trajectories are saved in the same PDB file or CIF file, PDB files can be imported locally into blender through the molecular nodes plug-in, but not the trajectory of the animation, all the frames overlap, the CIF file can not be imported, tried other software such as Mercury, ChimeraX, but no way!
Please, is there any way I can import these track animations saved in a single pdb or cif file into blender? Just like molecular nodes importing NMR animations directly from PDB website.
Extremely Grateful !

**Comment by BradyAJohnston** [comment]
Hi there. If your `.pdb` file has multiple different conformations of the same structure, Molecular Nodes should automatically import them as a trajectory. Are you able to provide more information about what is occurring, or maybe sharing any error messages / the files that you are trying to work with so I can have a look at them.

**Reply by lukun223344** [reply]
<img width="870" alt="test1" src="https://github.com/BradyAJohnston/MolecularNodes/assets/131518247/d2adf0f9-237d-43b3-b7ad-448cd199f79b">

[test_0_pX0_traj.zip](https://github.com/BradyAJohnston/MolecularNodes/files/13826559/test_0_pX0_traj.zip)

Thank you！

**Reply by BradyAJohnston** [reply]
Whichever tool is producing these `.pdb` files, is not properly producing them to spec. 

Each model requires `ENDMDL` at the end, and `MODEL` at the beginning. In this file they only have `ENDMDL`. 

If you insert `MODEL` at the beginning of each model, it imports correctly. 

This is down to the parsing that is done by `biotite`, and I don't think is something that I would end up changing unless there was a lot of demand for it (ideally people would use `cif` files instead as `pdb` is now deprectaed.

**Reply by lukun223344** [reply]
Yes, the changes for this single pdb file are not necessary, .cif files are definitely more generalized in the future; this PDB file was generated with RFdiffusion. Hopefully MN will add the ability to import .cif files directly.

**Reply by BradyAJohnston** [reply]
MN can already import `.cif` files. Is it not currently working for you?

**Reply by lukun223344** [reply]
[protein_trajectory.zip](https://github.com/BradyAJohnston/MolecularNodes/files/13826693/protein_trajectory.zip)
<img width="643" alt="test2" src="https://github.com/BradyAJohnston/MolecularNodes/assets/131518247/4b35f73c-a1c8-4b91-bd80-291ce747a6c4">
error

**Reply by BradyAJohnston** [reply]
This file is also not able to be read by ChimeraX, or MN. It is not a problem with MN, but rather a problem with a malformed `.cif` file. The `type_symbol` category (the element) is not present in the file. Whatever is writing the file is writing malformed files.

I can't help much with this, as without including the element information in the file you can't expect MN or other programs to be able to read the file.

**Reply by BradyAJohnston** [reply]
I suggest filing a bug report with the program that produces these files, so they can fix their file writing.

**Reply by lukun223344** [reply]
Got it. I'll mention this to them. Thanks again！

**Reply by lukun223344** [reply]
<img width="740" alt="test3" src="https://github.com/BradyAJohnston/MolecularNodes/assets/131518247/d463b2fd-6261-4bf9-a88c-0791e487676f">
The above PDB file has "MODEL X" added, so it can be imported normally now👍👍👍

**Reply by BradyAJohnston** [reply]
nice! glad to see that it's working :)
